pkg/doc/verifiable/example_credential_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+//UniversityDegree university degree
+type UniversityDegree struct {
+	Type       string `json:"type,omitempty"`
+	Name       string `json:"name,omitempty"`
+	College    string `json:"college,omitempty"`
+	University string `json:"university,omitempty"`
+}
+
+//UniversityDegreeSubject subject of university degree
+type UniversityDegreeSubject struct {
+	ID     string `json:"id,omitempty"`
+	Name   string `json:"name,omitempty"`
+	Spouse string `json:"spouse,omitempty"`
+
+	Degree UniversityDegree `json:"degree,omitempty"`
+}
+
+//UniversityDegreeCredential University Degree credential, from examples of https://w3c.github.io/vc-data-model
+type UniversityDegreeCredential struct {
+	c Credential
+
+	Subject *UniversityDegreeSubject `json:"credentialSubject,omitempty"`
+}
+
+func (udc *UniversityDegreeCredential) credential() *Credential {
+	return &udc.c
+}
+
+func (udc *UniversityDegreeCredential) decode(dataJSON []byte, credential *Credential) error {
+	return json.Unmarshal(dataJSON, udc)
+}
+
+func TestCredentialExtensibility(t *testing.T) {
+	udc := &UniversityDegreeCredential{}
+	cred, err := NewCredential(
+		[]byte(validCredential),
```

Comment:
We should add a utility function that would supply the credential type from the marshalled data. From there, someone could determine which type to instantiate.

Reply:
Created https://github.com/hyperledger/aries-framework-go/issues/218